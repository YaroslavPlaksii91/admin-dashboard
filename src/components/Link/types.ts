import { ReactNode } from 'react';

export type LinkProps = {
  url: string;
  children: ReactNode;
};
