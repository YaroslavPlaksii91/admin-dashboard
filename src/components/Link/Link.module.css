.link {
  font-size: 14px;
  font-weight: 600;
  line-height: 1.43;
  letter-spacing: 0.2px;
  color: var(--accent-blue-color);
}
