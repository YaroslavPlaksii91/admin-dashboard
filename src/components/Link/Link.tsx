import { FC } from 'react';
import { Link as RouterLink } from 'react-router-dom';

import { LinkProps } from './types';

import styles from './Link.module.css';

export const Link: FC<LinkProps> = ({ url, children }) => {
  return (
    <RouterLink to={url} className={styles.link}>
      {children}
    </RouterLink>
  );
};
