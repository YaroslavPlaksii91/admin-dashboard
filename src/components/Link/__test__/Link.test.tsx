import { MemoryRouter } from 'react-router-dom';
import { cleanup, render, screen } from '@testing-library/react';

import { Link } from '../Link';

describe('Link component', () => {
  afterEach(cleanup);

  const text = 'Link';
  const url = '/link';

  it('should be rendered', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Link text={text} url={url} />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be in the document', () => {
    render(
      <MemoryRouter>
        <Link text={text} url={url} />
      </MemoryRouter>,
    );

    const link = screen.getByRole('link');

    expect(link).toBeInTheDocument();
  });

  it('should have correct url address', () => {
    render(
      <MemoryRouter>
        <Link text={text} url={url} />
      </MemoryRouter>,
    );

    const link = screen.getByRole('link');

    expect(link).toHaveAttribute('href', url);
  });

  it('should have correct text', () => {
    render(
      <MemoryRouter>
        <Link text={text} url={url} />
      </MemoryRouter>,
    );

    const link = screen.getByText(text);

    expect(link).toBeInTheDocument();
  });
});
