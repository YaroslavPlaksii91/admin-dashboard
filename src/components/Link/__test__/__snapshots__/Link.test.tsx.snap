// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component should be rendered 1`] = `
<DocumentFragment>
  <a
    href="/link"
  >
    Link
  </a>
</DocumentFragment>
`;
